react/src/pages/MaintenancePage.tsx

```diff
@@ -1,89 +1,203 @@
+import { useTranslation } from 'react-i18next';
+import { StringParam, useQueryParam, withDefault } from 'use-query-params';
 
-const MaintenancePage: React.FC = () => {
+type TabKey = 'maintenance';
+type SettingGroup = {
+  title: string;
+  settingItems: SettingItemProps[];
+};
+
+const tabParam = withDefault(StringParam, 'maintenance');
+
+const MaintenanceSettingList = () => {
+  // state
+  const [isRecalculating, setIsRecalculating] = useState(false);
+  const [isRescanning, setIsRescanning] = useState(false);
+  // hooks
   const { t } = useTranslation();
+  const { upsertNotification } = useSetBAINotification();
+  const baiClient = useSuspendedBackendaiClient();
+  const painKiller = usePainKiller();
+  // funcs
+  const recalculateUsage = async () => {
+    setIsRecalculating(true);
 
-  const { value, dispatchEvent } = useWebComponentInfo();
-  let parsedValue: {
-    recalculating: boolean;
-    scanning: boolean;
-  };
-  try {
-    parsedValue = JSON.parse(value || '');
-  } catch (error) {
-    parsedValue = {
-      recalculating: false,
-      scanning: false,
-    };
-  }
-  const { recalculating, scanning } = parsedValue;
+    const task = new Promise((resolve, reject) => {
+      baiClient.maintenance
+        .recalculate_usage()
+        .then(() => resolve(null))
+        .catch((err: any) => {
+          console.error(err);
+          if (err && err.message) {
+            upsertNotification({
+              message: err.message,
+              description: painKiller.relieve(err.title),
+              open: true,
+            });
+          }
+          reject(err);
+        })
+        .finally(() => setIsRecalculating(false));
+    });
 
-  const { token } = theme.useToken();
+    upsertNotification({
+      message: t('maintenance.RecalculateUsage'),
+      description: t('maintenance.Recalculating'),
+      open: true,
+      backgroundTask: {
+        status: 'pending',
+        promise: task,
+        statusDescriptions: {
+          pending: t('maintenance.Recalculating'),
+          resolved: t('maintenance.RecalculationFinished'),
+          rejected: t('maintenance.RecalculationFailed'),
+        },
+      },
+    });
+  };
 
-  return (
-    <Flex
-      direction="row"
-      align="stretch"
-      style={{ margin: token.marginSM, gap: token.margin }}
-    >
-      <Row gutter={[14, token.margin]}>
-        <Col>
-          <Card title={t('maintenance.Fix')}>
-            <Typography.Title
-              level={5}
-              style={{
-                margin: 0,
-                paddingBottom: '12px',
-              }}
-            >
-              {t('maintenance.MatchDatabase')}
-            </Typography.Title>
-            <Trans>{t('maintenance.DescMatchDatabase')}</Trans>
+  const rescanImages = async () => {
+    setIsRescanning(true);
+    const notiKey = upsertNotification({
+      message: t('maintenance.rescanImages'),
+      description: t('maintenance.RescanImageScanning'),
```

Comment:
Key name has changed
```suggestion
      message: t('maintenance.RescanImages'),
```